backend/api/documents.py

```
@@ -0,0 +1,160 @@
+            )
+        await out_file.write(res.content)
+        await logger.adebug(
+            "Downloaded file at %s successfully.",
+            url,
+            out_filename=out_filename,
+        )
+
+
+async def download_github_project(url: str, dest_directory: os.PathLike):
+    """
+    Downloads a GitHub repository as a zip file and extracts it to the specified directory.
```

Ruff (0.11.9)

Line too long (91 > 80)

(E501)